extensions/virtual-threads/deployment/src/main/java/io/quarkus/virtual/threads/VirtualThreadsProcessor.java

```diff
@@ -0,0 +1,19 @@
+package io.quarkus.virtual.threads;
+
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
+import io.quarkus.deployment.builditem.ShutdownContextBuildItem;
+
+public class VirtualThreadsProcessor {
+
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
```

Comment:
I'm wondering if we can't move that to static.
It does not create thread, just prepare the executor.

It would be interesting to see how GraalVM reacts.

Reply:
VirtualThreads Executor is created at runtime. I don't think it'll change something for GraalVM. 
The config I am setting is a runtime config, but I could change it to a buildtime-runtime-fixed config.

btw I moved away from the enum introduced in the last RM PR because I am now shutting down the executor service.

Reply:
Static init and Runtime init are slightly different. Static init gives a change to Graalvm to inline the created object in the native executable. So, it does not need to be initialized during the runtime init. 

Because our scheduler does not depend on anything, it could work (if we make the shutdown timeout build time config)

Reply:
Ok switched to static init with build time config